File: pos-poc/pos_flutter/lib/data/models/employee.dart
import 'package:equatable/equatable.dart';

class Employee extends Equatable {
  final String id;
  final String tenantId;
  final String storeId;
  final String name;
  final String email;
  final String role;
  final String? pin;
  final bool isActive;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Employee({
    required this.id,
    required this.tenantId,
    required this.storeId,
    required this.name,
    required this.email,
    required this.role,
    this.pin,
    required this.isActive,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      id: json['id'] as String,
      tenantId: json['tenant_id'] as String,
      storeId: json['store_id'] as String,
      name: json['name'] as String,
      email: json['email'] as String,
      role: json['role'] as String,
      pin: json['pin'] as String?,
      isActive: json['is_active'] as bool,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'tenant_id': tenantId,
      'store_id': storeId,
      'name': name,
      'email': email,
      'role': role,
      'pin': pin,
      'is_active': isActive,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }

  @override
  List<Object?> get props => [
        id,
        tenantId,
        storeId,
        name,
        email,
        role,
        pin,
        isActive,
        createdAt,
        updatedAt,
      ];
}

File: pos-poc/pos-backend/internal/domain/employee.go
package domain

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

// EmployeeRole represents the role of an employee
type EmployeeRole string

const (
	RoleAdmin   EmployeeRole = "admin"
	RoleManager EmployeeRole = "manager"
	RoleCashier EmployeeRole = "cashier"
	RoleKitchen EmployeeRole = "kitchen"
)

// Employee represents an employee in the system
type Employee struct {
	ID           uuid.UUID      `gorm:"type:uuid;primary_key;default:uuid_generate_v4()" json:"id"`
	TenantID     uuid.UUID      `gorm:"type:uuid;not null" json:"tenant_id"`
	StoreID      *uuid.UUID     `gorm:"type:uuid" json:"store_id,omitempty"`
	Name         string         `gorm:"type:varchar(50);not null" json:"name"`
	Email        string         `gorm:"type:varchar(100)" json:"email,omitempty"`
	PinCode      string         `gorm:"type:varchar(10)" json:"pin_code,omitempty"`
	PasswordHash string         `gorm:"type:varchar(255)" json:"-"`
	Role         EmployeeRole   `gorm:"type:varchar(20);not null" json:"role"`
	Permissions  map[string]any `gorm:"type:jsonb" json:"permissions,omitempty"`
	IsActive     bool           `gorm:"default:true" json:"is_active"`
	CreatedAt    time.Time      `json:"created_at"`
	UpdatedAt    time.Time      `json:"updated_at"`

	// Relations
	Tenant *Tenant `gorm:"foreignKey:TenantID" json:"tenant,omitempty"`
	Store  *Store  `gorm:"foreignKey:StoreID" json:"store,omitempty"`
}

// TableName specifies the table name for Employee
func (Employee) TableName() string {
	return "employees"
}

// BeforeCreate hook
func (e *Employee) BeforeCreate(tx *gorm.DB) error {
	if e.ID == uuid.Nil {
		e.ID = uuid.New()
	}
	return nil
}
